toothpick-runtime/src/main/java/toothpick/ScopeImpl.java

```diff
@@ -78,6 +80,7 @@ public ScopeImpl(Object name) {
 
   @Override
   public <T> Lazy<T> getLazy(Class<T> clazz, String name) {
+    crashIfInvalid();
```

Comment:
We need to, it would get rid of the problem you mention when you say:

> There is one exception that won't fail:
When we use a closed scope to create an instance from a lazy (or provider) that does not contain any dependency (we are just creating the instance, but it does not need to inject anything).